### DISCUSSION TITLE
License options

### DISCUSSION THREAD

**Discussion by Ben-PH** [esp-rs/esp-template#93] [Q&A]
If we wanted to build a project under a GPL license (2 or 3), would that be okay, and what would we need to do? I'm not sure if the license being included in the template is a strict set of compatible licenses, or if it's just set up that way to, for example, reduce the likelihood of shoddy licensing of generated repos.

**Comment by jessebraham** [comment]
This is permitted by the Apache 2.0 license only for GPLv3:

> The [Free Software Foundation](http://www.fsf.org/) [considers](http://www.fsf.org/licensing/licenses/#apache2) the [Apache License, Version 2.0](https://www.apache.org/licenses/LICENSE-2.0) to be a [free software](http://www.gnu.org/philosophy/free-sw.html) license, [compatible](http://www.gnu.org/licenses/license-list.html#GPLCompatibleLicenses) with version 3 of the [GPL](http://www.fsf.org/licensing/licenses/gpl.html). The [Software Freedom Law Center](http://www.softwarefreedom.org/) provides [practical advice](http://www.softwarefreedom.org/resources/2007/gpl-non-gpl-collaboration.html) for developers about including permissively licensed source.
>
> Apache 2 software can therefore be included in GPLv3 projects, because the GPLv3 license accepts our software into GPLv3 works. However, GPLv3 software cannot be included in Apache projects. The licenses are incompatible in one direction only, and it is a result of ASF's licensing philosophy and the GPLv3 authors' interpretation of copyright law.

See [Apache License v2.0 and GPL Compatibility](https://www.apache.org/licenses/GPL-compatibility.html) for additional details.

You can see [LICENSE-APACHE](https://github.com/esp-rs/esp-template/blob/main/LICENSE-APACHE) in this repository for the full terms of compliance.

**Reply by Ben-PH** [reply]
So If I understand correctly, it's okay to remove the two license files, and any reference to licensing in the template-generated project, and replace it with GPL-3?
